### ISSUE TITLE
Rules for not exhaustive `match` case statements

### ISSUE THREAD

**Issue by crispyricepc** [astral-sh/ruff#20003]
### Summary

If a `match` statement is used with an enum type or some other type with a finite number of possible values, Ruff (or Ty, not sure which works better here) should warn about unused cases.

## Invalid code

```py
from enum import Enum

enum Animal(Enum):
    CAT = 0
    DOG = 1

animal = get_animal()

match animal: # warning should appear here
    case CAT:
        print("is a cat")
```

## Valid code

```py
from enum import Enum

enum Animal(Enum):
    CAT = 0
    DOG = 1

animal = get_animal()

match animal:
    case CAT:
        print("is a cat")
    case _:
        pass
```

This is inspired by a similar rule in pyright: `reportMatchNotExhaustive`

**Comment by MichaReiser** [comment]
This makes sense, but is something that is difficult for Ruff because it will require multifile analysis in most instances (and a good understanding of what values `animal` can expand to). 

That's why I'll move this issue to ty. It's a sub issue of https://github.com/astral-sh/ty/issues/887
